eth/downloader/queue.go

```diff
@@ -644,6 +645,8 @@ func (q *queue) expire(peer string, pendPool map[string]*fetchRequest, taskQueue
 	// Return any non-satisfied requests to the pool
 	if req.From > 0 {
 		taskQueue.Push(req.From, -int64(req.From))
+		// the invariant is that req.From > 0(for headers) and len(req.Headers) > 0(for bodies and receipts) can't both be true
+		return 1
```

Comment:
I don't understand the explanation. What is the reason for wanting to do a `return 1` here instead of leaving it be? What is the improvement?

If the driver for the change is "this is more correct" - as opposed to fixing some sort of malfunction, then I'd reject this.